[2025-01-29T03:47:24.482+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-01-29T03:47:24.488+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: etl_movies_pipeline.load_to_staging manual__2025-01-29T03:37:50.507097+00:00 [queued]>
[2025-01-29T03:47:24.491+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: etl_movies_pipeline.load_to_staging manual__2025-01-29T03:37:50.507097+00:00 [queued]>
[2025-01-29T03:47:24.491+0000] {taskinstance.py:2866} INFO - Starting attempt 3 of 3
[2025-01-29T03:47:24.554+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): load_to_staging> on 2025-01-29 03:37:50.507097+00:00
[2025-01-29T03:47:24.559+0000] {standard_task_runner.py:72} INFO - Started process 639 to run task
[2025-01-29T03:47:24.566+0000] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'etl_movies_pipeline', 'load_to_staging', 'manual__2025-01-29T03:37:50.507097+00:00', '--job-id', '22', '--raw', '--subdir', 'DAGS_FOLDER/etl_movies_pipeline.py', '--cfg-path', '/tmp/tmp2izspt4q']
[2025-01-29T03:47:24.568+0000] {standard_task_runner.py:105} INFO - Job 22: Subtask load_to_staging
[2025-01-29T03:47:24.596+0000] {task_command.py:467} INFO - Running <TaskInstance: etl_movies_pipeline.load_to_staging manual__2025-01-29T03:37:50.507097+00:00 [running]> on host 2fd33a817220
[2025-01-29T03:47:24.628+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='etl_movies_pipeline' AIRFLOW_CTX_TASK_ID='load_to_staging' AIRFLOW_CTX_EXECUTION_DATE='2025-01-29T03:37:50.507097+00:00' AIRFLOW_CTX_TRY_NUMBER='3' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-01-29T03:37:50.507097+00:00'
[2025-01-29T03:47:24.629+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-01-29T03:47:24.664+0000] {taskinstance.py:3311} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 422, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/etl_movies_pipeline.py", line 18, in load_to_staging
    connection = mysql.connector.connect(
                 ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/mysql/connector/pooling.py", line 322, in connect
    return CMySQLConnection(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/mysql/connector/connection_cext.py", line 151, in __init__
    self.connect(**kwargs)
  File "/home/airflow/.local/lib/python3.12/site-packages/mysql/connector/abstracts.py", line 1423, in connect
    self.config(**kwargs)
  File "/home/airflow/.local/lib/python3.12/site-packages/mysql/connector/abstracts.py", line 685, in config
    self.set_login(user, password)
  File "/home/airflow/.local/lib/python3.12/site-packages/mysql/connector/abstracts.py", line 1104, in set_login
    self._user = username.strip()
                 ^^^^^^^^^^^^^^
AttributeError: 'tuple' object has no attribute 'strip'
[2025-01-29T03:47:24.669+0000] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=etl_movies_pipeline, task_id=load_to_staging, run_id=manual__2025-01-29T03:37:50.507097+00:00, execution_date=20250129T033750, start_date=20250129T034724, end_date=20250129T034724
[2025-01-29T03:47:24.676+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-01-29T03:47:24.676+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 22 for task load_to_staging ('tuple' object has no attribute 'strip'; 639)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3005, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3159, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3183, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 422, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/etl_movies_pipeline.py", line 18, in load_to_staging
    connection = mysql.connector.connect(
                 ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/mysql/connector/pooling.py", line 322, in connect
    return CMySQLConnection(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/mysql/connector/connection_cext.py", line 151, in __init__
    self.connect(**kwargs)
  File "/home/airflow/.local/lib/python3.12/site-packages/mysql/connector/abstracts.py", line 1423, in connect
    self.config(**kwargs)
  File "/home/airflow/.local/lib/python3.12/site-packages/mysql/connector/abstracts.py", line 685, in config
    self.set_login(user, password)
  File "/home/airflow/.local/lib/python3.12/site-packages/mysql/connector/abstracts.py", line 1104, in set_login
    self._user = username.strip()
                 ^^^^^^^^^^^^^^
AttributeError: 'tuple' object has no attribute 'strip'
[2025-01-29T03:47:24.704+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-01-29T03:47:24.712+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-01-29T03:47:24.715+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
